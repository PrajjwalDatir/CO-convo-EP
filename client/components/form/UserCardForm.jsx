import React from 'react';
import PropTypes from 'prop-types';
import Button from '#/components/button';
import Card from '#/components/card';

function UserCardForm({
  header,
  alt,
  image,
  children,
  buttonText,
  underButton,
  onSubmit,
}) {
  return (
    <Card
      className="overflow-x-hidden border-solid border-blue-500 border-t-8"
      tabIndex="-1"
    >
      <div className="flex w-full">
        <div style={{ width: 300 }}>
          <h1 className="text-4xl font-light mb-4">{header}</h1>
          <form onSubmit={onSubmit}>
            {children}
            <Button
              className="w-full bg-blue-700 hover:bg-blue-800 text-gray-100"
              type="submit"
              overrideColors
            >
              {buttonText}
            </Button>
          </form>
          <div className="mt-3">{underButton}</div>
        </div>
        <div className="ml-6 -mr-3">
          <img className="w-64" alt={alt} src={image} />
        </div>
      </div>
    </Card>
  );
}

UserCardForm.propTypes = {
  header: PropTypes.string.isRequired,
  alt: PropTypes.string,
  image: PropTypes.string,
  children: PropTypes.node,
  buttonText: PropTypes.string,
  underButton: PropTypes.node,
  onSubmit: PropTypes.func.isRequired,
};

UserCardForm.defaultProps = {
  buttonText: 'Login',
  alt: '',
  image: '',
  underButton: null,
  children: [],
};

export default UserCardForm;
