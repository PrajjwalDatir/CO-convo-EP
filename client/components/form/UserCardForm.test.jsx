import React from 'react';
import { mount } from 'enzyme';
import * as R from 'ramda';
import UserCardForm from './UserCardForm';

describe('<UserCardForm />', () => {
  it('should match snapshot', () => {
    const wrapper = mount(
      <UserCardForm header="" buttonText="" onSubmit={R.always()} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the function passed in onSubmit', () => {
    const mock = jest.fn();
    const wrapper = mount(
      <UserCardForm header="" buttonText="" onSubmit={mock} />,
    );
    wrapper.find('button[type="submit"]').simulate('submit');
    expect(mock).toHaveBeenCalled();
  });

  // TODO
  it('should only take <LoginCardFormField /> as children', () => {
    const spy = jest.spyOn(console, 'warn').mockImplementation();
    mount(
      <UserCardForm header="" buttonText="" onSubmit={R.always()}>
        <h1>Not LoginCardFormField</h1>
      </UserCardForm>,
    );
    expect(spy).toHaveBeenCalled();
    spy.mockRestore();
  });
});
