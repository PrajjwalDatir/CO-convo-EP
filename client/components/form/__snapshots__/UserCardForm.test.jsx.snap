// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserCardForm /> should match snapshot 1`] = `
<UserCardForm
  alt=""
  buttonText=""
  header=""
  image=""
  onSubmit={[Function]}
  underButton={null}
>
  <Card
    className="overflow-x-hidden border-solid border-blue-500 border-t-4"
    tabIndex="-1"
  >
    <div
      className="rounded bg-white p-10 shadow-lg overflow-x-hidden border-solid border-blue-500 border-t-4"
      tabIndex="-1"
    >
      <div
        className="flex w-full"
      >
        <div
          style={
            Object {
              "width": 300,
            }
          }
        >
          <h1
            className="text-4xl font-light mb-4"
          />
          <form
            onSubmit={[Function]}
          >
            <Button
              className="w-full bg-blue-700 hover:bg-blue-800 text-gray-100"
              overrideColors={true}
              type="submit"
            >
              <button
                className="
        
        rounded focus:shadow-outline cursor-pointer px-4 py-2 shadow hover:shadow-md w-full bg-blue-700 hover:bg-blue-800 text-gray-100
      "
                type="submit"
              />
            </Button>
          </form>
          <div
            className="mt-3"
          />
        </div>
        <div
          className="ml-12 -mr-20"
        >
          <img
            alt=""
            className="w-64"
            src=""
          />
        </div>
      </div>
    </div>
  </Card>
</UserCardForm>
`;
